services/metadata_service/api/artifact.py

```diff
@@ -314,12 +322,16 @@ async def get_artifacts_by_step(self, request):
         artifacts = await self._async_table.get_artifact_in_steps(
             flow_name, run_number, step_name
         )
-
-        filtered_body = filter_artifacts_for_latest_attempt(
-            artifacts.body)
-        return web.Response(
-            status=artifacts.response_code, body=json.dumps(filtered_body)
-        )
+        if artifacts.response_code == 200:
+          filtered_body = filter_artifacts_for_latest_attempt(
```

Comment:
while we're at it, maybe add a type signature  for `filter_artifacts_for_latest_attempt()` for readability, for the next person who'll try to figure out what's going on

Reply:
I have added docstrings for those functions. Those can help understand the types for readability.